src/domdiv/main.py

```diff
@@ -1466,6 +1466,29 @@ def read_card_data(options):
 
 class CardSorter(object):
     def __init__(self, order, lang, baseCards):
+        # Set the locale to the selected language. Necessary for correct sorting using accented characters
```

Comment:
I don't totally understand how this works. I thought the local is a global state, so won't setting and restoring it in the constructor not affect anything as the sort() call that eventually calls the comparison happens later?

Reply:
The new temporary locale only surrounds the sort keys generation (lines 1493 - 1500), e.g.: `self.by_global_sort_key`. In `by_global_sort_key` definition I call `locale.strxfrm(card.name)`, which uses the temporarily set locale to create a string to one that can be used in locale-aware comparisons (see [`locale.strxfrm`](https://docs.python.org/3/library/locale.html#locale.strxfrm)). After all sort keys are generated, I restore the original locale.

This is inspired by the Example from [locale documentation](https://docs.python.org/3/library/locale.html#locale.CHAR_MAX).

I still cannot explain what causes the crashes on macOS. For instance, I noticed that the `locale.setlocale(locale.LC_COLLATE, ("es", "UTF-8"))` call _alone_ does not cause the crash: when I removed the subsequent `locale.strxfrm(card.name)` calls, the crash was gone. I attribute that to some concurrency issues or something. My understanding of the `locale` [doc](`locale.strxfrm(card.name)`) is that it is not so unexpected:

> The C standard defines the locale as a program-wide property that may be relatively expensive to change. On top of that, some implementations are broken in such a way that frequent locale changes may cause core dumps. This makes the locale somewhat painful to use correctly.

Unfortunately after a lot of searching, I do not think there is a better way of doing language correct alphabetical sort. [This article](https://realpython.com/python-sort-unicode-strings/) is a really good resource, but from what I gathered these are the only potential solutions:
- `pyuca` is quite old and does not work correctly for Czech (or potentially other languages)
- `PyICU` probably the best solution, but it is a Python extension which is quite difficult to set up (needs to be compiled on the target system)
- `locale.setlocale()` ...

Reply:
As far as I can see nothing is generated in the block you're surrounding by the local setting/restoring. We're simply assigning the functions to call during `sort(key=cardSorter)` via its `__call__` method. So at the time the actual sort key is generated, the local will already be restored. The right thing to do would probably be to write a decorator that sets/restores the locale and decorate all of the sort key methods with it.

Reply:
But then, not sure that won't cause crashes again and there are warnings all over the locale docs that this is not recommended.

Reply:
I reverted this change back (save current locale & restore back). Resolving this comment thread & will discuss my proposals in another comment